translation/src/main/scala/org/opencypher/gremlin/translation/ir/rewrite/GroupStepFilters.scala

```diff
@@ -93,6 +94,28 @@ object GroupStepFilters extends GremlinRewriter {
     traversals.flatMap {
       case SelectK(stepLabel) :: Values(propertyKey) :: Is(predicate) :: Nil =>
         (stepLabel, HasP(propertyKey, predicate)) :: Nil
+      case SelectK(stepLabel) :: ChooseP2(_, Id :: Nil) :: Is(_) :: Is(predicate) :: Nil =>
+        (stepLabel, HasP(T.id.getAccessor, predicate)) :: Nil
+      case Project(_ @_*) :: rest =>
+        val values = rest.map {
```

Comment:
This works, but we could make previous case to handle this case. Processing of within/without predicates [could be improved](https://github.com/dwitry/cypher-for-gremlin/commit/04b373bbae6d967abf3346db828a2a16a54d1c73) to provide cleaner Gremlin translation, that matches by `SelectK :: ChooseP2...` case you've added.

Reply:
Ok, I change it